### PR TITLE
Formatting for object descriptions.

### PR DESCRIPTION
**Pull request by DarviL82**
Allow descriptions of objects, such as Argument's or Command's to have richer information. At the moment, when printing the description on the help view, both the description specified to the argument and the one of the argument type are shown.
This is great, but it would be neat to allow more customization, such as being able to specify formatting semantics, like `{@link args.arg1}` or `{@link cmds.mySubcmd1}`.

## Example:
```java
Argument.create("test").description("Interesting argument. See also {@link args.arg1}.")
```

When being printed on the help, this is what would be shown:
```
--test:
   Interesting argument. See also --arg1.
```
`--arg1` would be shown with the color that represents it. And also, if it had multiple names, these would be shown like `--arg1/argument1`. Basically the argument representation that is normally shown for each argument.

## Formatters ideas:
- `@link`: Basically a "reference" to the specified element, as explained above.
- `@from`: Inlines the description of the selected element into the current one.

### Selectors:
After one of the formatters above is specified, a selector can be used to indicate the element to select (Command, argument, etc.):
- `args.<arg>`: The argument with the specified name.
  - `<arg>.type`: The argument type of the argument. At the moment the description of the type is automatically added. So with this change the user would need to add `@from !.type`. (`!` explained below).
- `cmds.<cmd>`: The command with the specified name.
  - Since a Command contains more arguments, this can be chained like `cmds.<cmd>.args.<arg>`.
- `groups.<grp>`: The group with the specified name.
- `!`: This selects the current element that is generating the description. For example, if the description of an argument named "foo" is set to `@link !`, this would be equal to `@link args.foo`.

## Syntax ideas:
- `@link:args.arg1`, suggested by @fadeoffical. No spaces required but possible problems when parsing if no spaces are present after it, like `desc: @link:args.arg1.`. The `.` could cause problems, but we could probably allow `{}` wrapping to indicate the formatting more explicitly.
- `@link args.arg1`, similar to javadocs syntax, though I personally prefer the former!
- `<link=args.arg1>`, much easier to parse with the eyes. Will go with this one.

### CODE REVIEW

**Comment by fadeoffical** [comment]
You sure you want to pull into main?

**Comment by DarviL82** [comment]
Nah it'll be merged into dev. Forgot to change that.

File: src/main/java/lanat/helpRepresentation/ArgumentRepr.java [hunk 1]
```diff
@@ -38,7 +53,7 @@ private ArgumentRepr() {}
 		return outText.toString();
 	}
 
-	public static @Nullable String getDescription(@NotNull Argument<?, ?> arg) {
+	public static @Nullable String getDescription(final @NotNull Argument<?, ?> arg) {
```
Comment (fadeoffical) [inline src/main/java/lanat/helpRepresentation/ArgumentRepr.java:56]: Is the final modifier really needed?
```suggestion
	public static @Nullable String getDescription(@NotNull Argument<?, ?> arg) {
```

File: src/main/java/lanat/helpRepresentation/ArgumentRepr.java [hunk 2]
```diff
@@ -47,7 +62,7 @@ private ArgumentRepr() {}
 		return ArgumentRepr.getRepresentation(arg) + ":\n" + HelpFormatter.indent(desc, arg);
 	}
 
-	static String getDescriptions(@NotNull List<@NotNull Argument<?, ?>> arguments) {
+	static String getDescriptions(final @NotNull List<@NotNull Argument<?, ?>> arguments) {
```
Comment (fadeoffical) [inline src/main/java/lanat/helpRepresentation/ArgumentRepr.java:65]: Is the final modifier really needed?
```suggestion
	static String getDescriptions(@NotNull List<@NotNull Argument<?, ?>> arguments) {
```

**Review by fadeoffical — CHANGES_REQUESTED** [review]
Lgtm! except the two finals.

**Pull request merged**